v3/how-to-guides/01-basics/c-constants/index.mdx

```diff
@@ -5,171 +5,145 @@ keywords: basics, runtime, configurable constants
-
-```rust
-#[pallet::hooks]
-impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
-  fn on_finalize(n: T::BlockNumber) {
-    if (n % T::ClearFrequency::get()).is_zero() {
-      let current_value = <SingleValue<T>>::get();
-      <SingleValue<T>>::put(0u32);
-      Self::deposit_event(Event::Cleared(current_value));
+
+Declaring a constant value in a runtime is a useful tool to either define fixed values or define values that change dynamically according to some factor. This guide shows you how to create pallet constants that are used to reset a \`u32\` value in storage. This value, we'll call \`SingleValue\`, can also be modified using a method called \`add_value\`.
+
+## Create pallet constants and add them to your runtime
+
+1. Define the constants in your pallet.
+
+  - `MaxAddend` will be the value displayed in metadata.
+  - `ClearFrequency` keeps track of the block numbers and will
+    be used to reset `SingleValue`:
+
+  ```rust
+  #[pallet::config]
+  pub trait Config: frame_system::Config {
+    type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+    
+    #[pallet::constant] // put the constant in metadata
+    /// Maximum amount added per invocation.
+    type MaxAddend: Get<u32>;
+
+    /// Frequency with which the stored value is deleted.
+    type ClearFrequency: Get<Self::BlockNumber>;
+    }
+  ```
+
+1. Declare your storage items and events.
+
+  Using the storage attribute macro, declare `SingleValue`, the value that gets modified every block cycle.
+
+  ```rust
+  #[pallet::storage]
+  #[pallet::getter(fn single_value)]
+  pub(super) type SingleValue<T: Config> = StorageValue<_, u32, ValueQuery>;
+  ```
+
+1. Define your pallet's events.
+
+  ```rust
+  #[pallet::event]
+  #[pallet::generate_deposit(pub(super) fn deposit_event)]
+  pub enum Event<T: Config> {
+      /// The value has been added to. The parameters are
+      /// (initial amount, amount added, final amount).
+      Added(u32, u32, u32),
+      /// The value has been cleared. The parameter is the value before clearing.
+      Cleared(u32)
+  }
+  ```
+
+1. Add an error that handles operation overflow:
+
+  ```rust
+  #[pallet::error]
+  pub enum Error<T> {
+    /// An operation would lead to an overflow.
+    Overflow
+  }
+  ```
+
+1. Configure `on_finalize`
+
+  `SingleValue` is set to 0 every `ClearFrequency` number of blocks in the `on_finalize` function that
+  runs at the end of block execution. Specify this logic under the `#[pallet::hooks]` attribute:
+
+  ```rust
+  #[pallet::hooks]
+  impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+    fn on_finalize(n: T::BlockNumber) {
+      if (n % T::ClearFrequency::get()).is_zero() {
+        let current_value = <SingleValue<T>>::get();
+        <SingleValue<T>>::put(0u32);
+        Self::deposit_event(Event::Cleared(current_value));
+      }
     }
   }
-}
-```
+  ```
 
-### 4. Create a method that allows users to specify the value.
```

Comment:
If you think my first suggestion above makes sense, I'd suggest adding: 

## Create a dispatchable and configure your runtime constants

here as a heading to the next two steps.